packages/headless/src/api/search/search-api-client.ts

```diff
@@ -29,15 +29,29 @@ import {HtmlRequest} from './html/html-request';
 import {findEncoding} from './encoding-finder';
 import {TextDecoder} from 'web-encoding';
 import {BaseParam} from '../platform-service-params';
-import {SearchThunkExtraArguments} from '../../app/search-thunk-extra-arguments';
 import {emptyQuestionAnswer} from '../../features/search/search-state';
 import {isNullOrUndefined} from '@coveo/bueno';
 import {AsyncThunkOptions} from '../../app/async-thunk-options';
+import {ClientThunkExtraArguments} from '../../app/thunk-extra-arguments';
+import {FacetSearchResponse} from './facet-search/facet-search-response';
+
+export interface FacetSearchAPIClient {
+  facetSearch(req: FacetSearchRequest): Promise<FacetSearchResponse>;
+}
 
 export type AllSearchAPIResponse = Plan | Search | QuerySuggest;
 
-export interface AsyncThunkSearchOptions<T extends Partial<SearchAppState>>
-  extends AsyncThunkOptions<T, SearchThunkExtraArguments> {
+export interface AsyncThunkSearchOptions<
+  T extends Partial<SearchAppState>
+> extends AsyncThunkOptions<
+    T,
+    ClientThunkExtraArguments<SearchAPIClient> & {
+      /*
+       * @deprecated This property is now unused, please use `apiClient` instead.
+       */
+      searchAPIClient?: SearchAPIClient;
```

Comment:
I thought it would make `searchApiClient` required here: https://github.com/coveo/ui-kit/pull/1259/files/a552687703b94284225d17eeef13bb04f11e596f#diff-af67e11dc46d216eda9fc554fb22ce64e15fa552abb56ddedbcc987efbd05c4eR80

but it is not! So let's keep put required :)